contracts/Wallet.sol

```diff
@@ -15,54 +16,87 @@ contract Wallet is IWallet {
-            IExtendedERC20 token = IExtendedERC20(erc20Tokens[i]);
-            paginatedInfo[count++] = TokenInfo({
-                tokenAddress: erc20Tokens[i],
+        if (tokenType == TokenType.ERC20) {
+            IERC20Metadata token = IERC20Metadata(tokenAddress);
+            uint256 balance = token.balanceOf(user);
+
+            // Filter tokens with a balance of 0 in ERC-20
+            if (balance > 0) {
+                info = TokenInfo({
+                    tokenAddress: tokenAddress,
+                    balance: token.balanceOf(user),
+                    name: token.name(),
+                    symbol: token.symbol(),
+                    decimals: token.decimals()
+                });
+            }
+
+        // Fill in ERC-721 token information
+        } else if (tokenType == TokenType.ERC721) {
+            IERC721Metadata token = IERC721Metadata(tokenAddress);
+            info = TokenInfo({
+                tokenAddress: tokenAddress,
                 balance: token.balanceOf(user),
                 name: token.name(),
                 symbol: token.symbol(),
-                decimals: token.decimals()
+                decimals: 0 // ERC721 tokens do not have decimals
             });
-        }
 
-        // Fill in ERC-721 token information
-        for (uint256 i = 0; i < erc721Tokens.length; i++) {
-            paginatedInfo[count++] = TokenInfo({
-                tokenAddress: erc721Tokens[i],
-                balance: 0,
-                name: "",
+        // Fill in ERC-1155 token information
+        } else if (tokenType == TokenType.ERC1155) {
+            IERC1155 token = IERC1155(tokenAddress);
+            // The token ID is needed to get the correct balance
+            // Assuming a token ID is provided or known
+            uint256 tokenId = 1; // Token ID needs to be determined, 1 is an example
+            info = TokenInfo({
+                tokenAddress: tokenAddress,
+                balance: token.balanceOf(user, tokenId),
+                name: "",   // Name and symbol are not standard in ERC1155
                 symbol: "",
-                decimals: 0
+                decimals: 0 // Decimals are not standard in ERC1155
             });
         }
 
-        // Fill in ERC-1155 token information
-        for (uint256 i = 0; i < erc1155Tokens.length; i++) {
-            paginatedInfo[count++] = TokenInfo({
-                tokenAddress: erc1155Tokens[i],
-                balance: 0, // Need tokenID
-                name: "",
-                symbol: "",
-                decimals: 0
-            });
+        return info;
+    }
+    
+    // Implement getPaginatedTokenInfo method
+    function getPaginatedTokenInfo(TokenType tokenType, address user, uint start, uint limit) 
+    external view override returns (TokenInfo[] memory tokensInfo){
+        address[] memory tokenAddresses = tokenListManager.getWhitelistedTokens(tokenType, start, limit);
```

Comment:
What if balance is 0 for some tokens, in which case the pagination view is incorrect.

Reply:
Solved

Reply:
The pagination view should not be applied to the whitelist, otherwise, the current page tokens may all of balance 0.

Reply:
Solved, I created a new function in TokenListmanager called getAllWhitelistedTokens. Get all tokens from whitelist first, then filter non-zero tokens.

Reply:
I add a function in TokenListManager called getAllWhitelistedTokens, but the logical structure is the same. First, all tokens are obtained from the whitelist, then tokens with 0 values are filtered out, and other non-zero tokens are displayed to the user. And I don't understand what means not be applied to the whitelist, otherwise, the current page tokens may all of balance 0.